File: janken_project/typescript/janken.ts
const readline = require('readline');

const choices = ["Rock", "Paper", "Scissors"];
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter 0 (Rock), 1 (Paper), 2 (Scissors): ", (answer: string) => {
    const userChoice = parseInt(answer);
    if (isNaN(userChoice) || userChoice < 0 || userChoice > 2) {
        console.log("Invalid input. Please enter 0, 1, or 2.");
        rl.close();
        return;
    }

    const computerChoice = Math.floor(Math.random() * 3);
    console.log(`You: ${choices[userChoice]}`);
    console.log(`Computer: ${choices[computerChoice]}`);

    if (userChoice === computerChoice) {
        console.log("It's a tie!");
    } else if ((userChoice === 0 && computerChoice === 2) ||
               (userChoice === 1 && computerChoice === 0) ||
               (userChoice === 2 && computerChoice === 1)) {
        console.log("You win!");
    } else {
        console.log("You lose!");
    }
    rl.close();
});

File: janken_project/kotlin/Janken.kt
fun main() {
  println("Enter 0 (Rock), 1 (Paper), 2 (Scissors): ")
  val userInput = readLine()
  val userChoice = userInput?.toIntOrNull()

  if (userChoice == null || userChoice !in 0..2) {
    println("Invalid input. Please enter 0, 1, or 2.")
    return
  }

  val computerChoice = (0..2).random()
  println("You: ${userChoice}")
  println("Computer: ${computerChoice}")

  when {
    userChoice == computerChoice -> println("It's a draw!")
    (userChoice == 0 && computerChoice == 2) || (userChoice == 1 && computerChoice == 0) || (userChoice == 2 && computerChoice == 1) -> println("You win!")
    else -> println("You lose!")
  }
}